pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -1042,8 +1073,11 @@ in
     nvimRequireCheck = "git-worktree";
   };
 
+  guard-collection = super.guard-nvim.overrideAttrs {
+    dependencies = with self; [ guard-nvim ];
+  };
+
   guard-nvim = super.guard-nvim.overrideAttrs {
-    dependencies = with self; [ guard-collection ];
     nvimRequireCheck = "guard";
   };
```

Comment:
Why was this change made? guard.nvim depends on guard-collection: https://github.com/nvimdev/guard.nvim?tab=readme-ov-file#usage

Reply:
> -- Builtin configuration, optional
and guard-collection actually depends on guard-nvim. It can't be in both places otherwise you get a circular dependency.

Reply:
Turns out `guard-collection` only depends on `guard-nvim` in tests (do we have something like `nativeCheckInputs` for neovim plugins?).

Almost all if not all people who use this plugin, use `guard-collection`, and who didn't specify it as a dependency will get a slightly annoying warning, for example *[Guard]: "yapf": needs nvimdev/guard-collection to access builtin configuration*.

However I do understand that it is technically an optional dependency and the correct thing is to not specify as a strict dependency, thanks for pointing that out :)

Also:
```diff
-  guard-collection = super.guard-nvim.overrideAttrs {
+  guard-collection = super.guard-collection.overrideAttrs {
```

Reply:
> Turns out `guard-collection` only depends on `guard-nvim` in tests (do we have something like `nativeCheckInputs` for neovim plugins?).
> 
> Almost all if not all people who use this plugin, use `guard-collection`, and who didn't specify it as a dependency will get a slightly annoying warning, for example _[Guard]: "yapf": needs nvimdev/guard-collection to access builtin configuration_.
> 
> However I do understand that it is technically an optional dependency and the correct thing is to not specify as a strict dependency, thanks for pointing that out :)
> 
> Also:
> 
> ```diff
> -  guard-collection = super.guard-nvim.overrideAttrs {
> +  guard-collection = super.guard-collection.overrideAttrs {
> ```

Thanks, I can fix that and yes I can add it to nativeCheckInputs